documentation/src/main/asciidoc/topics/upgrading.adoc

```diff
@@ -65,6 +65,21 @@ provided configuration. The possible statuses of the cache health are now HEALTH
 
 The Kryo and Protostuff marshallers have been deprecated and will be removed in Infinispan 15.0
 
+=== {brandname} includes an upgraded version of the ProtoStream API that can
+affect upgrade from previous {brandname} versions.
+
+Protobuf message types that you generate with annotation processing at compile
+time must have the same nesting in your annotated Java classes. If the nesting
+in your generated messages does not match the nesting in your classes, then
+data incompatibility issues can occur.
```

Comment:
I think we can remove this paragraph. The subsequent paragraph explains the issue and what needs to be done, this just confuses me tbh.